# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

Makes the code compile as C++17. std::random_shuffle was removed in C++17.

diff --git packages/epetraext/src/coloring/EpetraExt_MapColoring.cpp packages/epetraext/src/coloring/EpetraExt_MapColoring.cpp
index d16eb5df662..09acbdaaa8a 100644
--- packages/epetraext/src/coloring/EpetraExt_MapColoring.cpp
+++ packages/epetraext/src/coloring/EpetraExt_MapColoring.cpp
@@ -57,6 +57,7 @@
 #include <Epetra_Time.h>
 
 #include <vector>
+#include <random>
 #include <set>
 #include <map>
 
@@ -199,7 +200,9 @@ Toperator( OriginalTypeRef orig  )
       for( int i = 0; i < nCols; ++i )
         rowOrder[ i ] = i;
 #ifndef TFLOP
-      std::random_shuffle( rowOrder.begin(), rowOrder.end() );
+      std::random_device rng;
+      std::mt19937 urng( rng() );
+      std::shuffle( rowOrder.begin(), rowOrder.end(), urng );
 #endif
     }
 
@@ -609,7 +612,9 @@ Toperator( OriginalTypeRef orig  )
       for( int i = 0; i < nRows; ++i )
         rowOrder[i] = i;
 #ifdef TFLOP
-      random_shuffle( rowOrder.begin(), rowOrder.end() );
+      std::random_device rng;
+      std::mt19937 urng( rng() );
+      std::shuffle( rowOrder.begin(), rowOrder.end(), urng );
 #endif
     }
 
